File: http.py
import urllib
import httplib
import re
import StringIO
import sys
import time

sesid = 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'

regexp = re.compile(r'<div id="question">(.+)</div>')
regexp_h = re.compile(r'<input type="hidden" name="h_timestamp" value="(.+)" />')
regexp_i = re.compile(r'<input type="hidden" name="i_timestamp" value="(.+)" />')
headers = {
'Content-Type':'application/x-www-form-urlencoded',
'Cookie':'PHPSESSID='+sesid+';'
};

connection = httplib.HTTPConnection('www.hackerschool.jp')
#connection = httplib.HTTPConnection('1.33.188.154')
sttm0 = time.time()
connection.request('GET','/hack/take32.php',None,headers)
sttm1 = time.time()
response = connection.getresponse();
sttm2 = time.time()
print response.status,response.reason
if response.status == 200:
	data = response.read()
	res = regexp.search(data)
	res_h = regexp_h.search(data)
	res_i = regexp_i.search(data)
	arr = res.group(1).split("x")
	answer = 1;
	for elm in arr:
		answer = answer * int(elm.strip())
	edtm = time.time()
	connection.close()
	connection2 = httplib.HTTPConnection('www.hackerschool.jp')
	#connection2 = httplib.HTTPConnection('1.33.188.154')
	#connection2 = httplib.HTTPConnection('localhost',8080)
	for index in range(100):
		sttm0 = time.time()
		connection2.request('POST','/hack/take32.php','h_timestamp='+res_h.group(1)+'&i_timestamp='+res_i.group(1)+'&input_id='+str(answer),headers)
		#connection2.request('POST','http://www.hackerschool.jp/hack/take32.php','h_timestamp='+res_h.group(1)+'&i_timestamp='+res_i.group(1)+'&input_id='+str(answer),headers)
		sttm1 = time.time()
		response = connection2.getresponse();
		edtm = time.time()
		print response.status,response.reason
		if response.status == 200:
			data = response.read()
			res = regexp.search(data)
			res_h = regexp_h.search(data)
			res_i = regexp_i.search(data)
			arr = res.group(1).split("x")
			answer = 1;
			for elm in arr:
				answer = answer * int(elm.strip())
			edtm = time.time()
			#data = response.read()
			print data
			buf = StringIO.StringIO(data)
			print '-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-'
			lines = buf.readlines()
			showcount = 0
			for ln in lines:
				if ln.find('<div class="control-group ">') >= 0:
					showcount = 10
				if showcount > 0:
					sys.stdout.write(ln)
					showcount = showcount - 1
			# print res.group(1) + ',' + res_h.group(1) + ','  + res_i.group(1)
			#connection2.close()
			print str(edtm - sttm0)
			print str(edtm - sttm1)
			# time.sleep(1)
	

File: http.rb
#!/usr/bin/ruby
# encoding: utf-8
require 'rubygems'
require 'mechanize'
require 'kconv'

sesid = 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'

regexp = Regexp.compile('<div id="question">(.+)</div>')
headers = 
{'Accept'=>'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
'Origin'=>'http://www.hackerschool.jp',
'Content-Type'=>'application/x-www-form-urlencoded',
'User-Agent'=>'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_6_8) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/31.0.1650.63 Safari/537.36',
'Referer'=>'http://www.hackerschool.jp/hack/take32.php',
'Accept-Encoding'=>'sdch',
'Accept-Language'=>'ja,en-US;q=0.8,en;q=0.6',
'Cookie'=>'PHPSESSID='+sesid+';'
};
uri = 'http://www.hackerschool.jp/hack/take32.php';
_uri = URI.parse(uri)
agent = Mechanize.new
cookie_string = 'PHPSESSID='+sesid+'; domain=www.hackerschool.jp; expires=Thursday, 14-Oct-#{Time.new.year+1} 13:20:05 GMT; path=/'
Mechanize::Cookie.parse(_uri, cookie_string){|cookie|
  agent.cookie_jar.add(_uri, cookie)
}
agent.max_file_buffer = 4611686018427388000
page = agent.get(uri, headers=headers)
puts page.code
if page.code == '200' then
	sttm = Time.now
	data = page.at('#question')
	regexp.match(data.to_s)
	buf = $1
	arrbuf = buf.split('x');
	answer = 1
	for elm in arrbuf do
		answer *= Integer(elm.strip)
	end
	for index in 1...1000 do
		form = page.form_with(:name=>'form1')
		form.field_with(:name=>'input_id').value=answer.to_s
		button = form.button_with(:value=>'送信！')
		page = agent.submit(form,button)
		if page.code == '200' then
			edtm = Time.now
#			puts page.body
			puts '-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-'
			puts page.at('.control-group')
			puts (edtm - sttm).to_s
			puts answer

			sttm = Time.now
			data = page.at('#question')
			regexp.match(data.to_s)
			buf = $1
			arrbuf = buf.split('x');
			answer = 1
			for elm in arrbuf do
				answer *= Integer(elm.strip)
			end
		end
	end
end
